Chapel 1.20 changes

jesunsahariar opened the pull request:
Changes we made to CHGL to be compatible with Chapel v1.20

jesunsahariar committed:
Changing WorkQueue to be compatible with Chapel 1.2

jesunsahariar committed:
Fixing errors for different test cases.

jesunsahariar committed:
Modification of the initializers so as to ensure using cyclic distributions for vertices and edges.

LouisJenkinsCS commented:
We need Chapel 1.20 integration testing first (I.E green check before this gets merged)

jesunsahariar committed:
Updating travis to point to Chapel 1.20.0

jesunsahariar committed:
Updating travis to point to multi-locale Chapel.

jesunsahariar committed:
For the butterfly test, ake a graph argument instead

As per Louis, verticesDomain is accessing the field directly via `this`.verticesDomain where `this` is remote.
`this` is the class instance of the original locale that invoked the method
And when the algo jump to a remote locale, accessing this explicitly or implicitly results in communication.
CylicDist  is optimized to disable locality checks when accessing the DSI methods directly, like we are doing
We use graph.verticesDomain instead it will resolve it through privatization (I.E through record wrapper forwarding).

jesunsahariar committed:
We needed to grab privatized instance of the hypergraph and use that
for accessing getEdges in a forall iterator (since they are parallel and distributed)
in GraphTest.

jesunsahariar committed:
Fixed errors in ActiveDNS code due to upgrade.
